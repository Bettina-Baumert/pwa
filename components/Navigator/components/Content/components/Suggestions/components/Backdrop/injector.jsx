import React from 'react';
import { NavigatorContext } from '../../../../../../context';

export default Component => () => (
  <NavigatorContext>
    {({ toggleSearchField }) => (
      <Component toggleSearchField={toggleSearchField} />
    )}
  </NavigatorContext>
);

