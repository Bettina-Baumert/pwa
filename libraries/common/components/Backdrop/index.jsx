import React, { Component } from 'react';
import { css } from 'glamor';
import PropTypes from 'prop-types';
import style from './style';

/**
 * Backdrop component.
 */
class Backdrop extends Component {
  /**
   * The component prop types.
   * @type {Object}
   */
  static propTypes = {
    className: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.shape(),
    ]),
    color: PropTypes.string,
    isVisible: PropTypes.bool,
    level: PropTypes.number,
    onClick: PropTypes.func,
    opacity: PropTypes.number,
  };

  /**
   * The component default props.
   * @type {Object}
   */
  static defaultProps = {
    className: {},
    color: '#000',
    isVisible: false,
    level: 2,
    onClick: () => {},
    opacity: 50,
  };

  /**
   * Only update when the `isVisible` prop changes.
   * @param {Object} nextProps The next set of component props.
   * @return {boolean}
   */
  shouldComponentUpdate(nextProps) {
    return (this.props.isVisible !== nextProps.isVisible);
  }

  /**
   * @returns {string}
   */
  get className() {
    return css(style, this.props.className);
  }

  /**
   * @returns {Object}
   */
  get style() {
    const { color, level, opacity } = this.props;

    return {
      opacity: (opacity / 100),
      background: color,
      zIndex: level,
    };
  }

  /**
   * Renders the component.
   * @returns {JSX}
   */
  render() {
    const { isVisible } = this.props;

    if (!isVisible) {
      return null;
    }

    return (
      <div aria-hidden style={this.style} className={this.className} onClick={this.props.onClick} />
    );
  }
}

export default Backdrop;
