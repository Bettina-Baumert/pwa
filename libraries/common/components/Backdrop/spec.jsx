import React from 'react';
import { shallow } from 'enzyme';
import Backdrop from './index';

describe('<Backdrop />', () => {
  let mockOpen;

  beforeEach(() => {
    mockOpen = jest.fn();
  });

  it('should render', () => {
    const wrapper = shallow(<Backdrop isVisible />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should execute callback when Backdrop is clicked', () => {
    const wrapper = shallow(<Backdrop isVisible onClick={mockOpen} />);
    wrapper.find('div').simulate('click');
    expect(mockOpen).toBeCalled();
  });
});
