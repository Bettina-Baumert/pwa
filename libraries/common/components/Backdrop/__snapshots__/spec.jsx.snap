// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> should render 1`] = `
<div
  aria-hidden={true}
  className={
    Object {
      "data-css-hlk66h": "",
    }
  }
  onClick={[Function]}
  style={
    Object {
      "background": "#000",
      "opacity": 0.5,
      "zIndex": 2,
    }
  }
/>
`;
